app/models/user.rb

```diff
@@ -107,6 +93,43 @@ def decorate
     UserDecorator.new(self)
   end
 
+  def max_login_attempts?
+    second_factor_attempts_count.to_i >= max_login_attempts
+  end
+
+  def max_login_attempts
+    3
+  end
+
+  def create_direct_otp
+    update(
+      direct_otp: random_base10(TwoFactorAuthenticatable.direct_otp_length),
+      direct_otp_sent_at: Time.zone.now.utc,
```

Comment:
`Time.zone` will be UTC, so the `.utc` at the end is redundant

```suggestion
      direct_otp_sent_at: Time.zone.now,
```